commit 53e724067a4ee9373972079e225d0d5f683b9c5a
Author: Marc Zyngier <maz@kernel.org>
Date:   Wed Jan 23 13:21:58 2013 -0500

    ARM: KVM: arch_timers: Add guest timer core support
    
    Add some the architected timer related infrastructure, and support timer
    interrupt injection, which can happen as a resultof three possible
    events:
    
    - The virtual timer interrupt has fired while we were still
      executing the guest
    - The timer interrupt hasn't fired, but it expired while we
      were doing the world switch
    - A hrtimer we programmed earlier has fired
    
    Reviewed-by: Will Deacon <will.deacon@arm.com>
    Signed-off-by: Christoffer Dall <c.dall@virtualopensystems.com>
    Signed-off-by: Marc Zyngier <marc.zyngier@arm.com>

diff --git a/arch/arm/kvm/interrupts.S b/arch/arm/kvm/interrupts.S
index 5f113bedfaee..8ca87ab0919d 100644
--- a/arch/arm/kvm/interrupts.S
+++ b/arch/arm/kvm/interrupts.S
@@ -95,6 +95,7 @@ ENTRY(__kvm_vcpu_run)
 	save_host_regs
 
 	restore_vgic_state
+	restore_timer_state
 
 	@ Store hardware CP15 state and load guest state
 	read_cp15_state store_to_vcpu = 0
@@ -189,6 +190,7 @@ after_vfp_restore:
 	read_cp15_state store_to_vcpu = 1
 	write_cp15_state read_from_vcpu = 0
 
+	save_timer_state
 	save_vgic_state
 
 	restore_host_regs